src/game/GameManager.cpp

```diff
@@ -251,7 +302,7 @@ std::shared_ptr<Weapon> GameManager::getWeapon(const int32_t id){
 void GameManager::deleteWeaponDrop(const int32_t id) {
 
     const auto& it = weaponDropManager.find(id);
-    if(it != weaponDropManager.end()){
+    if(it != weaponDropManager.end()) {
```

Comment:
Call to find isn't required
This way creates a temporary, and since you are deleting based on the provided id, the temp iterator isn't used
Remove it and replace if condition with call to `.count`